langserver/cmd/langserver-go/langserver-go.go

```diff
@@ -66,6 +67,9 @@ func run() error {
 			jsonrpc2.NewConn(context.Background(), jsonrpc2.NewBufferedStream(conn, jsonrpc2.VSCodeObjectCodec{}), langserver.NewHandler(), connOpt...)
 		}
 
+	case "ws":
+		return modes.WebSocket(*addr, connOpt)
```

Comment:
why is this a library now in this version of the diff? before it was quite a simple and small code snippet. I assume I should hold off reviewing until you fix up this PR to exclude code you didn't mean to include?